src/engine/Input/Pointer.ts

```diff
@@ -223,8 +223,8 @@ export class Pointer extends Class {
    * Checks if Pointer has a specific Actor in ActorsUnderPointer list.
    * @param actor An Actor for check;
    */
-  public isActorUnderPointer(actor: Actor): boolean {
-    return this._actorsUnderPointer.hasOwnProperty(actor.id.toString());
+  public isActorAliveUnderPointer(actor: Actor): boolean {
+    return !!(!actor.isKilled() && actor.scene && this._actorsUnderPointer.hasOwnProperty(actor.id.toString()));
```

Comment:
It'll be good to go through the codebase and replace these dicts with Sets and Maps now they we have more efficient structures.

Reply:
Do Sets and Maps downlevel well in TypeScript?